# Stock X Data Contest for the data nerds

Your submission should consist of a single pdf file with the following items:

1 chart or graph displaying your results. Do not exceed 1 chart
A short write-up describing your analysis/results. Do not exceed 200 words
One entry per participant
Make sure the pdf has your name and email address somewhere on it

![StockX image](https://s3.amazonaws.com/stockx-sneaker-analysis/wp-content/uploads/2019/02/StockX_Data-Contest_Logo1-1100x1118.jpg "stockX logo")

In [10]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from scipy import stats
import plotly.plotly as py
import plotly.graph_objs as go
import plotly.tools as tls
import colorlover as cl

#####################################################
#####################################################

                ##### STOCK X #####
                     # DATA #
                #### CHALLENGE ####
            
#####################################################
#####################################################

path = 'https://s3.amazonaws.com/stockx-sneaker-analysis/wp-content/uploads/2019/02/StockX-Data-Contest-2019.xlsx'

df = pd.read_excel(path, sheet_name=1)

df.head()

Order Date   Brand                                   Sneaker Name  \
0 2017-09-01   Yeezy           Adidas-Yeezy-Boost-350-Low-V2-Beluga   
1 2017-09-01   Yeezy    Adidas-Yeezy-Boost-350-V2-Core-Black-Copper   
2 2017-09-01   Yeezy     Adidas-Yeezy-Boost-350-V2-Core-Black-Green   
3 2017-09-01   Yeezy       Adidas-Yeezy-Boost-350-V2-Core-Black-Red   
4 2017-09-01   Yeezy  Adidas-Yeezy-Boost-350-V2-Core-Black-Red-2017   

   Sale Price  Retail Price Release Date  Shoe Size  Buyer Region  
0      1097.0           220   2016-09-24       11.0    California  
1       685.0           220   2016-11-23       11.0    California  
2       690.0           220   2016-11-23       11.0    California  
3      1075.0           220   2016-11-23       11.5      Kentucky  
4       828.0           220   2017-02-11       11.0  Rhode Island

In [9]:
type(df)

pandas.core.frame.DataFrame

In [11]:
df.corr()

Sale Price  Retail Price  Shoe Size
Sale Price      1.000000     -0.361549   0.082795
Retail Price   -0.361549      1.000000  -0.052798
Shoe Size       0.082795     -0.052798   1.000000

## From the blog post 
The data we’re giving you consists of a random sample of all Off-White x Nike and Yeezy 350 sales from between 9/1/2017 (the month that Off-White first debuted “The Ten” collection) and the present. There are 99,956 total sales in the data set; 27,794 Off-White sales, and 72,162 Yeezy sales. The sample consists of U.S. sales only.

To create this sample, we took a random, fixed percentage of StockX sales (X%) for each colorway, on each day, since September 2017. So, for each day the Off-White Jordan 1 was on the market, we randomly selected X% of its sale from each day. (It’s not important to know what X is; all that matters is that it’s a random sample, and that the same fixed X% of sales was selected from every day, for every sneaker).

We’ve included 8 variables for you to work with: Order Date, Brand, Sneaker Name, Sale Price ($), Retail Price ($), Release Date, Shoe Size, and Buyer State (the U.S. state the buyer shipped to). You can use whatever variables you want in the analysis; you can use 1 variable, or you can use all 8. And remember, every row in the spreadsheet represents an individual StockX sale. There are no averages or order counts; this is just a random sample of daily sales data

In [25]:
## Creating data frames for yeezys and offwhite shoes

off_white = df.copy()
off_white = off_white[off_white['Brand'].isin(['Off-White'])]

yeezy = df.copy()
yeezy = yeezy[yeezy['Brand'].isin([' Yeezy'])]

print('Shape of Off-White dataframe : {}'.format(off_white.shape))
print('Shape of Yeezy dataframe : {}'.format(yeezy.shape))

Shape of Off-White dataframe : (27794, 8)
Shape of Yeezy dataframe : (72162, 8)


In [23]:
#lmao this dataframe is only yeezys and offwhite i dig it
df['Brand'].unique()

array([' Yeezy', 'Off-White'], dtype=object)

# Ideas for graphs/charts/trends to explore

- Region/Map of where the most buyers are for yeezys vs. offwhite sneakers
- and what does the this trend ^ explain about the economy in that state or how much disposable income people have to be spending $1,000 on a pair of sneakers LOL
- Another trend: what sneakers are more popular? Kind of obvious because there are almost 3X as many yeezys sold than offwhite in the entire dataset
- 